### ISSUE TITLE
TFT - How to Define Different Encoder Lengths for Each Time Series

### ISSUE THREAD

**Issue by fg1985** [jdb78/pytorch-forecasting#229]
Hello,

Is it possible to define a different encoder length for each time-series (with TFT model)? if so how would you do that?
What I am trying to do is to create a model that takes as input several time-series and generates a prediction to only one of them, I want the model to take as input different lookback windows for each time-series (for example, for time-series_1 encode 10 last values, and for time-series_2 encode 100 last values)

I have one more question, for example, let's assume I have one time-series that represents the number of cars on the road at each timestep (each hour), I'm willing to create a model that will forecast every day's last hour traffic (number of cars on the road) according to the traffic in the first few hours of the day, the next input and forecast will be of the next day, is there a way to configure the prediction horizon to a single value but with steps other than 1? (predicting the next hour but only every X hours)  - basically that is the functionality of this method  https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/timeseries_dataset_from_array

Thanks!

**Comment by jdb78** [comment]
I think this is not natively possible. HOWEVER, you can manually create an index of all subsequences in the dataset. This is internally done by creating the "index" attribute of the dataset. You could filter it to satisfy your needs. The corresponding data is stored in the attribute "data". I imagine it could be useful in the future to have an extra indicator column in the data frame for data points to be either encoder only, decoder only or either. Would be a great PR!

**Comment by fg1985** [comment]
Thank you for the answer!
Regarding my Second question, do you think there is a way to define a stride other than 1 from one prediction to another (also in the training phase)?

Thanks again, 
Guy

**Comment by jdb78** [comment]
No yet. But you can always modify the index.

**Comment by fg1985** [comment]
Thanks!
I have managed to modify the stride by changing the code of __iter__ method in TimeSynchronizedBatchSampler Class:

`def __iter__(self):
        if self.shuffle:  # shuffle samples
            groups = {name: shuffle(group) for name, group in self._groups.items()}
        else:
            groups = self._groups

        batch_samples = np.random.permutation(len(self))
        
        stride = 10
        batch_samples.sort()
        for idx in np.arange(0, len(batch_samples),stride):
            name = self._group_index[idx]
            sub_group = self._sub_group_index[idx]
            sub_group_start = sub_group * self.batch_size
            sub_group_end = sub_group_start + self.batch_size
            batch = groups[name][sub_group_start:sub_group_end]
            yield batch`

**Comment by fg1985** [comment]
# @jdb78 
I tried multiple ways to manipulate the index attribute in order to get a different encoder length for each time series (group_id) but without success, I suppose the needed changes should be made at the _construct_index method, any chance you can offer me some guidance on how to do that?
Thanks again!

**Comment by sairamtvv** [comment]
@veds12  Any luck on changing the encoder length for each time series
